test/src/com/xilinx/rapidwright/eco/TestECOTools.java

```diff
@@ -416,6 +423,53 @@ public void testCreateCell() {
         }
     }
 
+    @Test
+    public void testCreateInlineCellOnInputPin() {
+        Design d = new Design("Test", Device.KCU105);
+
+        Cell and2 = d.createAndPlaceCell("and2", Unisim.AND2, "SLICE_X100Y100/A6LUT");
+        Cell button0 = d.createAndPlaceIOB("button0", PinType.IN, "AE10", "LVCMOS18");
+        Cell button1 = d.createAndPlaceIOB("button1", PinType.IN, "AF9", "LVCMOS18");
+        Cell led0 = d.createAndPlaceIOB("led0", PinType.OUT, "AP8", "LVCMOS18");
+
+        // Connect Button 0 to the LUT2 input I0
+        EDIFHierCellInst hierButton0 = button0.getEDIFHierCellInst().getParent();
+        Net net0 = d.createNet(new EDIFHierNet(hierButton0, hierButton0.getCellType().getNet("O")));
+        ECOTools.connectNet(d, and2, "I0", net0);
+
+        // Connect Button 1 to the LUT2 input I1
+        EDIFHierCellInst hierButton1 = button1.getEDIFHierCellInst().getParent();
+        Net net1 = d.createNet(new EDIFHierNet(hierButton1, hierButton1.getCellType().getNet("O")));
+        ECOTools.connectNet(d, and2, "I1", net1);
+
+        // Connect the LUT2 (AND2) to the LED IO
+        Net net2 = d.createNet("and2");
+        net2.connect(and2, "O");
+        net2.connect(led0, "I");
+
+        // Route site internal nets
+        d.routeSites();
+
+        // Insert a LUT1 in between 'and2.I0' and its source, 'button0.O'
+        EDIFHierPortInst input = and2.getEDIFHierCellInst().getPortInst("I0");
+        Site site = d.getDevice().getSite("SLICE_X100Y101");
+        BEL bel = site.getBEL("A6LUT");
+        Unisim lut1Type = Unisim.LUT1;
+        ECOTools.createInlineCellOnInputPin(d, input, lut1Type, site, bel, "I0", "O");
+
+        // Route nets between sites
+        new Router(d).routeDesign();
+
+        Cell lut1 = d.getSiteInstFromSite(site).getCell(bel);
+        Assertions.assertNotNull(lut1);
+        Assertions.assertEquals(lut1Type.name(), lut1.getEDIFHierCellInst().getCellType().getName());
+        Assertions.assertEquals(net0, lut1.getSitePinFromLogicalPin("I0", null).getNet());
+        Assertions.assertNotEquals(net0, lut1.getSitePinFromLogicalPin("O", null).getNet());
+
+        Assumptions.assumeTrue(FileTools.isVivadoOnPath());
```

Comment:
It looks like failing assumptions anywhere in the test causes the whole test to be skipped.

Instead, check Vivado is on the path using something like:
https://github.com/Xilinx/RapidWright/blob/ede179cf44d968bbb7bcb25abf281826b8353aff/test/src/com/xilinx/rapidwright/design/TestCell.java#L109-L111

Or more recently, I've been using `TestRWRoute.assertVivadoFullyRouted()` which is a NOP if no Vivado.